layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -1350,10 +1350,43 @@ the right."
   (interactive)
   (set-buffer-file-coding-system 'undecided-dos nil))
 
-(defun spacemacs/copy-file ()
-  "Write the file under new name."
-  (interactive)
-  (call-interactively 'write-file))
+(defun spacemacs/copy-file (arg)
+  "Write the file under new name and select action on new file.
+This command prompts for a new filename then prompts for
+selecting an action to perform on the new file.
+
+If prefixed with the universal ARG \\[universal-argument] the
+prompt includes the buffer file name.
+
+When prompted for selecting an action, pressing `?' will open the
+help menu that explains the different options."
+  (interactive "P")
+  (let* ((new-file-name (read-file-name "Save file as: "
+                                        (if arg buffer-file-name default-directory))))
+    ;; when selecting a directory name, add (buffer-name) or message error for
+    ;; non-visiting file buffers
+    (when (directory-name-p new-file-name)
+      (if buffer-file-name
+          (setq new-file-name (concat new-file-name (buffer-name)))
+        (error "Copying of non-visiting file buffers requires to specify new file name.")))
+    (when (and (file-exists-p new-file-name)
+               (not (y-or-n-p (format "File %s exists. Overwrite?: " new-file-name))))
+      (error "Canceled"))
+    (pcase
+        (read-answer "New file action: "
```

Comment:
In general avoid string comparison. This is inefficient.

You can just match the character code.

If you really want, match symbol, e.g. 'only-write.

Reply:
Thanks for this info, handy to be aware of. But in this case the code only runs once only when this command is called, so I guess the efficiency is not so important in this case.

So otherwise you would suggest to compare only the first (or other relevant) characters or otherwise compare with `intern`?

Reply:
Ah yes, I did not know about `user-error`. Indeed I used a catch-throw before, but I thought `error` looks nicer (and then accept the "wrong" behavior). Very useful info!

Reply:
> So otherwise you would suggest to compare only the first (or other relevant) characters, or alternatively compare by converting to symbol with `intern`?

See https://www.gnu.org/software/emacs/manual/html_node/elisp/Multiple-Queries.html

- wrap this in a let form to set read-answer-short to t. 
- Change the long answer to a single character string. the long answer is not shown but is returned. Since it has length 1, it's fast to compare.
- Since read-answer-short is t, user only need to type one key to select a choice, instead of being required to type a sentence/phrase

Reply:
Yes I was aware of this, but then I would probably have wrapped it in `read-answer-short 'auto` (and indeed make the long answers slightly shorter, but still a little more informative)